purchase_landed_cost/models/purchase_cost_distribution.py

```diff
@@ -0,0 +1,603 @@
+        comodel_name='purchase.order.line', string='Purchase order line',
+        related='move_id.purchase_line_id')
+    purchase_id = fields.Many2one(
+        comodel_name='purchase.order', string='Purchase order', readonly=True,
+        related='move_id.purchase_line_id.order_id', store=True)
+    partner = fields.Many2one(
+        comodel_name='res.partner', string='Supplier', readonly=True,
+        related='move_id.purchase_line_id.order_id.partner_id')
+    picking_id = fields.Many2one(
+        'stock.picking', string='Picking', related='move_id.picking_id',
+        store=True)
+    product_id = fields.Many2one(
+        comodel_name='product.product', string='Product', store=True,
+        compute='_compute_product_id')
+    product_qty = fields.Float(
+        string='Quantity', compute='_get_product_qty', store=True)
+    product_uom = fields.Many2one(
+        comodel_name='uom.uom', string='Unit of measure',
+        related='move_id.product_uom')
+    product_price_unit = fields.Float(
+        string='Unit price', related='move_id.price_unit')
+    expense_lines = fields.One2many(
+        comodel_name='purchase.cost.distribution.line.expense',
+        inverse_name='distribution_line', string='Expenses distribution lines')
+    product_volume = fields.Float(
+        string='Volume', help="The volume in m3.",
+        related='product_id.product_tmpl_id.volume')
+    product_weight = fields.Float(
+        string='Gross weight', related='product_id.product_tmpl_id.weight',
+        help="The gross weight in Kg.")
+    standard_price_old = fields.Float(
+        string='Previous cost', compute="_compute_standard_price_old",
+        store=True,
+        digits=dp.get_precision('Product Price'))
+    expense_amount = fields.Float(
+        string='Cost amount', digits=dp.get_precision('Account'),
+        compute='_compute_expense_amount')
+    cost_ratio = fields.Float(
+        string='Unit cost', compute='_compute_cost_ratio')
+    standard_price_new = fields.Float(
+        string='New cost', digits=dp.get_precision('Product Price'),
+        compute='_compute_standard_price_new')
+    total_amount = fields.Float(
+        compute=_compute_total_amount, string='Amount line',
+        digits=dp.get_precision('Account'))
+    total_weight = fields.Float(
+        compute=_compute_total_weight, string="Line weight", store=True,
+        digits=dp.get_precision('Stock Weight'),
+        help="The line gross weight in Kg.")
+    total_volume = fields.Float(
+        compute=_compute_total_volume, string='Line volume', store=True,
+        help="The line volume in m3.")
+    company_id = fields.Many2one(
+        comodel_name="res.company", related="distribution.company_id",
+        store=True,
+    )
+
+    @api.model
+    def get_action_purchase_cost_distribution(self):
+        xml_id = 'purchase_landed_cost.action_purchase_cost_distribution'
+        action = self.env.ref(xml_id, raise_if_not_found=True).read()[0]
```

Comment:
True is the default value for raise_if_not_found:
https://github.com/odoo/odoo/blob/36551615b7265ad83dbf9d1207ff9ee59b7069f3/odoo/api.py#L792
```suggestion
        action = self.env.ref(xml_id).read()[0]
```

Reply:
Right, change done